### PR TITLE
Vector image support for imagetoVTK

### PR DESCRIPTION
**Pull request by JonTong**

### CODE REVIEW

**Comment by somada141** [comment]
@renefritze what do you think?

File: pyevtk/hl.py [hunk 1]
```diff
@@ -100,11 +101,17 @@ def imageToVTK(path, origin = (0.0,0.0,0.0), spacing = (1.0,1.0,1.0), cellData =
     if cellData != None:
         keys = list(cellData.keys())
         data = cellData[keys[0]]
-        end = data.shape
+        if hasattr(data,'shape'):
+            end = data.shape
+        elif(data[0].ndim==3 and data[1].ndim==3 and data[2].ndim==3):
```
Comment (somada141) [inline pyevtk/hl.py]: been forever since I've looked at this code but doesn't this assume 3D vector arrays?

Comment (JonTong) [reply pyevtk/hl.py]: I guess it does - I only thought about using it for 3D vector fields like velocity etc.

Would it be worth generalising it to higher dimensional vectors?

Comment (somada141) [reply pyevtk/hl.py]: I think it should support whatever the format supports but I've long passed the reigns to @renefritze so I'll let him decide

**Comment by renefritze** [comment]
So the question is, @JonTong would you want to add the general case?

**Comment by JonTong** [comment]
I'm happy to look at adding the general case in the near future.
To clarify, N-d vectors in 3D space rather than a full generalisation of both the number of components in the main vector (N) and the dimensional space of each component (m) (N component vectors in m-D space), right?

**Comment by renefritze** [comment]
> I'm happy to look at adding the general case in the near future.
Great!
> To clarify, N-d vectors in 3D space rather than a full generalisation of both the number of components in the main vector (N) and the dimensional space of each component (m) (N component vectors in m-D space), right?
TBH, I can't really say, since I don't have a use case. Just do what's useful for you first I'd say.

**Comment by somada141** [comment]
I reckon @renefritze is right, lets merge this in and should people want the generalised version we can revisit.

**Pull request merged**